bin/pygrb/pycbc_pygrb_efficiency

```diff
@@ -219,62 +227,65 @@ offsource_trigs.reverse()
 max_bestnr, _, full_time_veto_max_bestnr =\
     ppu.max_median_stat(slide_dict, time_veto_max_bestnr, trig_bestnr, total_trials)
 
+# ==========================
+# Calculate template chirp masses from bank
+# ==========================
+logging.info('Reading template chirp masses')
+with h5py.File(opts.bank_file, 'r') as bank_file:
+    template_mchirps = mchirp_from_mass1_mass2(
+        bank_file['mass1'][:],
+        bank_file['mass2'][:]
+    )
+
+
 
 # =======================
 # Load on source triggers
 # =======================
 if onsource_file:
 
     # Get trigs
-    on_trigs = ppu.load_xml_table(onsource_file, "multi_inspiral")
-    logging.info("%d onsource triggers loaded.", len(on_trigs))
+    on_trigs = ppu.load_triggers(onsource_file, vetoes)
+    logging.info("%d onsource triggers loaded.", on_trigs['network/event_id'].len())
 
-    # Separate off chirp mass column
-    on_mchirp = on_trigs.get_column('mchirp')
+    # Calculate chirp mass values
+    on_mchirp = template_mchirps[on_trigs['network/template_id']]
 
     # Set loudest event arrays
-    #loud_on_bestnr_trigs = None
     loud_on_bestnr = 0
-    #loud_on_fap = 1
 
-    # Calculate BestNRs and record loudest trig by BestNR
+    # Retrieve BestNRs and record loudest trig by BestNR.
+    # Get indices of loudest triggers and pick the loudest.
     if on_trigs:
-        on_trigs_bestnrs = ppu.get_bestnrs(on_trigs,
-                                           q=chisq_index, n=chisq_nhigh,
-                                           null_thresh=null_thresh,
-                                           snr_threshold=snr_thresh,
-                                           sngl_snr_threshold=sngl_snr_thresh,
-                                           chisq_threshold=new_snr_thresh,
-                                           null_grad_thresh=null_grad_thresh,
-                                           null_grad_val=null_grad_val)
-        loud_on_bestnr_trigs, loud_on_bestnr = \
-            np.asarray([[x, y] for y, x in sorted(zip(on_trigs_bestnrs, on_trigs),
-                                                  key=lambda elem: elem[0],
-                                                  reverse=True)])[0]
+        loud_on_bestnr_idx = np.argsort(on_trigs['network/reweighted_snr'])
+        loud_on_bestnr_idx = np.flip(loud_on_bestnr_idx)
```

Comment:
I'd prefer storing only the first element only of the right hand side.  While it is irrelevant, `loud_on_bestnr` stores a single value and it would be cleaner to do the same with `loud_on_bestnr_idx`